src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -182,16 +182,21 @@ protected function isCircularReference($object, &$context)
      * If a circular reference handler is set, it will be called. Otherwise, a
      * {@class CircularReferenceException} will be thrown.
      *
-     * @param object $object
+     * @param object      $object
+     * @param string|null $format
+     * @param array       $context
```

Comment:
all these can be removed imho

Reply:
It allows the IDE to discover those parameters. They could be dropped only when/if we'll uncomment the new parameters.

Reply:
ok, then next question for me is: how will we make these real args in 5.0?
shouldn't we throw a deprecation somehow?

Reply:
do we want them to be optional in 5.x ? I would say no (otherwise you would not be able to rely on them in the handler).

so I would trigger a deprecation in case they are not provided when calling the method.

and I suggest making this method `@final`, as the proper way to extend the handling of circular references is to register a handler, not to extend the method in child classes.

Reply:
> making this method @final

if that fits, then that's the best, as just adding the annotation will trigger a deprecation notice automatically

Reply:
+1 for final